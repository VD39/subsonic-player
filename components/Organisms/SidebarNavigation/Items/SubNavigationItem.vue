<script setup lang="ts">
import ButtonLink from '@/components/Atoms/ButtonLink.vue';

defineProps<{
  collapsed: boolean;
  icon: Icon;
  title: string;
  to: string;
}>();
</script>

<template>
  <li>
    <ButtonLink
      is="nuxt-link"
      :to="to"
      :icon="icon"
      class="sidebarLink"
      exact-active-class="sidebarCurrentLink"
      :show-text="!collapsed"
      :title="title"
    >
      {{ title }}
    </ButtonLink>
  </li>
</template>
