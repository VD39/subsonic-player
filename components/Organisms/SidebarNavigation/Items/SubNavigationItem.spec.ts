import type { VueWrapper } from '@vue/test-utils';

import { mount } from '@vue/test-utils';

import SubNavigationItem from './SubNavigationItem.vue';

function factory(props = {}) {
  return mount(SubNavigationItem, {
    props: {
      collapsed: false,
      icon: ICONS.add,
      title: 'title',
      to: 'to',
      ...props,
    },
  });
}

describe('SubNavigationItem', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
