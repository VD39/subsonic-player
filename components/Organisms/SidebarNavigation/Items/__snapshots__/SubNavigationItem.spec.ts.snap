// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubNavigationItem > matches the snapshot 1`] = `
"<li><a exactactiveclass="sidebarCurrentLink" class="centerItems buttonLink sidebarLink sidebarLink" title="title">
    <phpluscircle class="icon" aria-hidden="true" size="24" weight="regular"></phpluscircle><span class="clamp text">title</span>
  </a></li>"
`;
